odoo/models.py

```diff
@@ -2054,7 +2054,11 @@ def _read_group_process_groupby(self, gb, query):
             }
             if tz_convert:
                 qualified_field = "timezone('%s', timezone('UTC',%s))" % (self._context.get('tz', 'UTC'), qualified_field)
-            qualified_field = "date_trunc('%s', %s::timestamp)" % (gb_function or 'month', qualified_field)
+            week_start = babel.core.Locale.parse(get_lang(self.env).code).first_week_day
+            # 0 is monday, 6 is sunday
+            week_start = 0 if gb_function != 'week' else (7 - week_start) % 7
```

Comment:
That seems significantly less readable as an inline expression. It also exponentially increases the complexity of the `qualified_field` expression even in the "nominal" case where `gb_function` is not the week.

Why not have a normal conditional where:

- if the `gb_function` is not `'week'` we proceed as before
- if it is we can do all the weird adjustment stuff

That avoids unnecessarily paying for all the adjustments in the common case where we don't need it.